src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/MessageReadinessConsumer.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.wacaseeventhandler.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.wacaseeventhandler.clients.LaunchDarklyFeatureFlagProvider;
+import uk.gov.hmcts.reform.wacaseeventhandler.config.features.FeatureFlag;
+import uk.gov.hmcts.reform.wacaseeventhandler.entity.CaseEventMessageEntity;
+import uk.gov.hmcts.reform.wacaseeventhandler.entity.MessageState;
+import uk.gov.hmcts.reform.wacaseeventhandler.repository.CaseEventMessageRepository;
+import uk.gov.hmcts.reform.wacaseeventhandler.util.UserIdParser;
+
+import java.util.List;
+
+@Slf4j
+@Component
+@SuppressWarnings("PMD.DoNotUseThreads")
+@ConditionalOnProperty("azure.servicebus.enableASB")
+public class MessageReadinessConsumer implements Runnable {
+
+    private final DeadLetterQueuePeekService deadLetterQueuePeekService;
+    private final CaseEventMessageRepository caseEventMessageRepository;
+    private final LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    public MessageReadinessConsumer(DeadLetterQueuePeekService deadLetterQueuePeekService,
+                                    CaseEventMessageRepository caseEventMessageRepository,
+                                    LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider) {
+        this.deadLetterQueuePeekService = deadLetterQueuePeekService;
+        this.caseEventMessageRepository = caseEventMessageRepository;
+        this.launchDarklyFeatureFlagProvider = launchDarklyFeatureFlagProvider;
+    }
+
+    @Override
+    @Transactional
+    public void run() {
+        final List<CaseEventMessageEntity> allMessageInNewState = caseEventMessageRepository.getAllMessagesInNewState();
+
+        // TODO Alternatively could get a list of message Ids of messages containing userId and pass
+        // that to bulk update method
+        allMessageInNewState.stream()
+                .filter(msg -> launchDarklyFeatureFlagProvider
+                        .getBooleanValue(FeatureFlag.DLQ_DB_PROCESS, getUserId(msg)))
+                .forEach(this::checkMessageToMoveToReadyState);
+    }
+
+    private void checkMessageToMoveToReadyState(CaseEventMessageEntity messageInNewState) {
+        if (deadLetterQueuePeekService.isDeadLetterQueueEmpty()) {
```

Comment:
can we add unit test to cover when this is not empty?

Reply:
Fixed in https://github.com/hmcts/wa-case-event-handler/pull/218/commits/24188312c85cd5b24bba2b8c69e1d909286b7e97